lib/writer.rb

```diff
@@ -0,0 +1,32 @@
+require 'json'
+
+class Writer
+
+  def initialize(path, file = nil)
+    @path = path
+    @file = file
+  end
+
+  def run(contacts)
+    file.nil? ? write(contacts) : tests_write(contacts)
+  end
+
+  private
+
+  attr_reader :path, :file
+
+  def parse(contacts)
+    contacts.to_json
+  end
+
+  def write(contacts)
+    File.open(path, "w+") do |file|
```

Comment:
If you inject file, you should find it easier to test. Also, this block variable shadows the file attr_reader which makes this class more difficult to reason about.

Explore the rubydocs for the classes you're using (File / IO in this case), there are other ways to force the file to be written to disk: http://ruby-doc.org/core-2.2.0/IO.html#method-i-flush or http://ruby-doc.org/core-2.2.0/IO.html#method-i-close.